### PR TITLE
Support proper capsule.jvm.args parsing

### PR DESCRIPTION
**Pull request by aalberti**
Capsule now parses capsule.jvm.args as command line arguments.
It escapes quoted, double quoted parameters,
and characters escaped by \

### CODE REVIEW

File: capsule/src/main/java/Capsule.java [hunk 1]
```diff
@@ -55,20 +55,8 @@
...
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.EnumSet;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (aalberti) [inline capsule/src/main/java/Capsule.java]: Oupsie, didn't see that. Should I fix it?

File: capsule/src/main/java/Capsule.java [hunk 2]
```diff
@@ -2470,6 +2454,65 @@ private void resolveNativeDependencies() {
         return new ArrayList<String>(jvmArgs.values());
     }
 
+    private List<String> parseCommandLineArguments(String toProcess) {
```
Comment (aalberti) [inline capsule/src/main/java/Capsule.java]: How do you want me to state that this is heavily inspired by http://grepcode.com/file/repo1.maven.org/maven2/org.apache.ant/ant/1.9.5/org/apache/tools/ant/types/Commandline.java#Commandline.translateCommandline%28java.lang.String%29 ?

File: capsule/src/test/java/CapsuleTest.java [hunk 3]
```diff
@@ -1040,6 +1040,29 @@ public void testWrapperCapsuleNonCapsuleApp() throws Exception {
...
+        Jar jar = newCapsuleJar()
+                .setAttribute("Main-Class", "MyCapsule")
+                .setAttribute("Premain-Class", "MyCapsule")
+                .setAttribute("Application-Class", "com.acme.Foo")
+                .addClass(MyCapsule.class);
+
+        List<String> args = list();
+        List<String> cmdLine = list();
+
+        System.setProperty("capsule.jvm.args", "-Ddouble.quoted.arg=\"escape me\" -Dsingle.quoted.arg='escape me' -Dspace.escaped.arg=escape\\ me");
```
Comment (aalberti) [inline capsule/src/test/java/CapsuleTest.java]: I couldn't find a better way than System.setProperty to set capsule.jvm.args. Do you have a suggestion?
In addition, I'm now realizing I didn't reset the property after the test :/ At least I'll do this if you don't have a better solution.

File: capsule/src/main/java/Capsule.java (hunk 1 above)
Comment (pron) [inline capsule/src/main/java/Capsule.java]: Yes, please :)

File: capsule/src/main/java/Capsule.java (hunk 2 above)
Comment (pron) [inline capsule/src/main/java/Capsule.java]: A comment would do. Write something like: "this code is a modified version of ..."

File: capsule/src/test/java/CapsuleTest.java (hunk 3 above)
Comment (pron) [inline capsule/src/test/java/CapsuleTest.java]: Take a look at `setProperties(klass, props)` used throughout the test.

**Comment by aalberti** [comment]
OK, I'll do that. In the meantime, I realized I left the company git script that adds a ChangeId to all commit messages (for gerrit). Do you prefer me to add commits with this change id, of rebase the whole chain?

**Comment by pron** [comment]
Cool, thanks! Does it work on Windows? Do you have any reason to suspect why it might not?

**Comment by aalberti** [comment]
You're right, using \ as an escape char is not a good idea on windows... I'm working on it. I might revert this whole addition to ant's cmd line parser, if I don't find a simple clever idea.

**Comment by aalberti** [comment]
Unless you have additional comment, I'm considering it done.... and if you agree, I'm gonna need a 1.0.3 release ;)

**Comment by aalberti** [comment]
Just quickly tested our product with capsule 1.0.3-SNAPSHOT, and it works fine for me. When are you planning a 1.0.3 release?

**Comment by pron** [comment]
Next week.

**Comment by aalberti** [comment]
Thanks a lot. Looking forward to it

**Comment by pron** [comment]
Released.

**Pull request merged**